### DISCUSSION TITLE
Autodiscovery by IMEI and IMSI doesn' t work !

### DISCUSSION THREAD

**Discussion by CpServiceSpb** [RoEdAl/asterisk-chan-quectel#6] [General]
I use quectel channel with Asterisk 20.04 on Ubuntu 22.04 x64 LTS.
There are some Huawe 3G dongles as following ongle channel is loaded.
Autodiscovery for Huawei dongles is on and works.

Autodiscover for Quectel is also on and correct IMEI and IMSI are spceified but it doesn' t work.

I read your comments, in particular regardinig:

> Using modern serial port locking methods:
> 
>     ioctl(fd, TIOCGEXCL, &locking_status) and ioctl(fd, TIOCEXCL),
>     flock(fd, LOCK_EX | LOCK_NB)

But as I understand quectel channel goes throught each ttyUSB... port to make attempting to identify itself on ports.

I wpuld like to recommend for this to diable locking mode.
Remain locking mode only for connecting to identified port.

I can not exactly say if there is a reason that autodsicovery doesn' t work or not.
But it doesn' t work.
Please correct it.

**Comment by RoEdAl** [comment]
My plans are rather to remove autodiscovery completely. Autodiscovery code is very buggy (and unreadable) and I don't have time to improve or rewrite it.

**Reply by CpServiceSpb** [reply]
It depends on dongle/module model.
For example, autodiscover works mostly fine with Huawei E1550 and Huawei E180.

The main problem without autodiscovery is mutable ttyUSB devices after next or next by next rebooting of OS or after removing / adding some other USB modems.

I wrote for testing purposes the soft wch ssign symlonks with human reaadable names for ttyUSBs by imei or even IMSI.
But it is required more tests especially for imsi.
If you want I will send it to tes.

**Reply by CpServiceSpb** [reply]
And chan-quectel works with symlink.
But if you launch Asterisk from asterisk user, you have to set up appropriate owner and/or group during symlink creation.
But it is also autodiscover but via other soft.

**Comment by RoEdAl** [comment]
On systemd-based systems symlinks of serial devices are created into `/dev/serial/by-id` and `/dev/serial/by-path` directories by default. For *OpenWRT* you may try my [persistent-tty](http://github.com/RoEdAl/persistent-tty) script. These symlinks are sufficient for me.

**Reply by CpServiceSpb** [reply]
My soft allows to do o by imei or ven imsi.
